File: mobile_apps/ios_app/TigerWallet/Screens/HomeScreen.swift
import SwiftUI

// Home Screen - Main Dashboard
struct HomeScreen: View {
    @State private var selectedTab: String = "Wallet"
    @State private var totalBalance: Double = 12450.00
    
    let tabs = ["Wallet", "DApps", "Activity"]
    
    let tokens = [
        TokenItem(symbol: "ETH", name: "Ethereum", balance: 4.2, value: 8400.00, icon: "◈"),
        TokenItem(symbol: "USDT", name: "Tether", balance: 2500.00, value: 2500.00, icon: "₮"),
        TokenItem(symbol: "BNB", name: "BNB", balance: 1.5, value: 450.00, icon: "🟡"),
        TokenItem(symbol: "SOL", name: "Solana", balance: 22.0, value: 1100.00, icon: "☀️")
    ]
    
    var body: some View {
        TabView(selection: $selectedTab) {
            walletView
                .tabItem { Label("Wallet", systemImage: "wallet.pass") }
                .tag("Wallet")
            
            dappsView
                .tabItem { Label("DApps", systemImage: "app.badge") }
                .tag("DApps")
            
            activityView
                .tabItem { Label("Activity", systemImage: "clock.arrow.circlepath") }
                .tag("Activity")
        }
    }
    
    var walletView: some View {
        NavigationView {
            ScrollView {
                VStack(spacing: 20) {
                    // Balance Card
                    balanceCard
                    
                    // Quick Actions
                    quickActions
                    
                    // Token List
                    tokenList
                }
                .padding()
            }
            .navigationTitle("TigerWallet")
            .toolbar {
                ToolbarItem(placement: .navigationBarTrailing) {
                    Button(action: {}) {
                        Image(systemName: "bell")
                    }
                }
            }
        }
    }
    
    var balanceCard: some View {
        VStack(spacing: 16) {
            VStack(spacing: 4) {
                Text("Total Balance")
                    .font(.subheadline)
                    .foregroundColor(.secondary)
                Text("$\(totalBalance, specifier: "%.2f")")
                    .font(.system(size: 36, weight: .bold))
            }
            
            HStack(spacing: 20) {
                Button(action: {}) {
                    HStack {
                        Image(systemName: "arrow.up.circle.fill")
                        Text("Send")
                    }
                    .padding(.horizontal, 20)
                    .padding(.vertical, 10)
                    .background(Color.orange)
                    .foregroundColor(.white)
                    .cornerRadius(20)
                }
                
                Button(action: {}) {
                    HStack {
                        Image(systemName: "arrow.down.circle.fill")
                        Text("Receive")
                    }
                    .padding(.horizontal, 20)
                    .padding(.vertical, 10)
                    .background(Color.gray.opacity(0.2))
                    .foregroundColor(.primary)
                    .cornerRadius(20)
                }
            }
        }
        .padding(24)
        .background(
            LinearGradient(
                gradient: Gradient(colors: [Color.orange.opacity(0.8), Color.orange]),
                startPoint: .topLeading,
                endPoint: .bottomTrailing
            )
        )
        .foregroundColor(.white)
        .cornerRadius(16)
    }
    
    var quickActions: some View {
        HStack(spacing: 16) {
            quickActionButton(icon: "arrow.triangle.2.circlepath", label: "Swap", color: .blue)
            quickActionButton(icon: "bridge.fill", label: "Bridge", color: .purple)
            quickActionButton(icon: "chart.line.uptrend.xyaxis", label: "Stake", color: .green)
            quickActionButton(icon: "photo.on.rectangle", label: "NFTs", color: .pink)
        }
    }
    
    func quickActionButton(icon: String, label: String, color: Color) -> some View {
        VStack(spacing: 8) {
            Button(action: {}) {
                Image(systemName: icon)
                    .font(.title2)
                    .frame(width: 50, height: 50)
                    .background(color.opacity(0.2))
                    .foregroundColor(color)
                    .cornerRadius(12)
            }
            Text(label)
                .font(.caption)
                .foregroundColor(.secondary)
        }
    }
    
    var tokenList: some View {
        VStack(alignment: .leading, spacing: 12) {
            HStack {
                Text("Assets")
                    .font(.headline)
                Spacer()
                Button(action: {}) {
                    Text("Add Token")
                        .font(.caption)
                        .foregroundColor(.orange)
                }
            }
            
            ForEach(tokens) { token in
                tokenRow(token: token)
            }
        }
    }
    
    func tokenRow(token: TokenItem) -> some View {
        HStack(spacing: 12) {
            // Icon
            ZStack {
                Circle()
                    .fill(Color.gray.opacity(0.2))
                    .frame(width: 44, height: 44)
                Text(token.icon)
                    .font(.title3)
            }
            
            // Name & Symbol
            VStack(alignment: .leading, spacing: 2) {
                Text(token.name)
                    .font(.subheadline)
                    .fontWeight(.medium)
                Text(token.symbol)
                    .font(.caption)
                    .foregroundColor(.secondary)
            }
            
            Spacer()
            
            // Balance & Value
            VStack(alignment: .trailing, spacing: 2) {
                Text("\(token.balance, specifier: "%.4f")")
                    .font(.subheadline)
                    .fontWeight(.medium)
                Text("$\(token.value, specifier: "%.2f")")
                    .font(.caption)
                    .foregroundColor(.secondary)
            }
        }
        .padding()
        .background(Color.gray.opacity(0.1))
        .cornerRadius(12)
    }
    
    var dappsView: some View {
        NavigationView {
            ScrollView {
                VStack(alignment: .leading, spacing: 20) {
                    // Featured DApps
                    VStack(alignment: .leading, spacing: 12) {
                        Text("Featured")
                            .font(.headline)
                        
                        ScrollView(.horizontal, showsIndicators: false) {
                            HStack(spacing: 12) {
                                dappCard(name: "Uniswap", category: "DeFi", icon: "🦄")
                                dappCard(name: "OpenSea", category: "NFT", icon: "🌊")
                                dappCard(name: "Aave", category: "Lending", icon: "👻")
                            }
                        }
                    }
                    
                    // All DApps
                    VStack(alignment: .leading, spacing: 12) {
                        Text("All DApps")
                            .font(.headline)
                        
                        ForEach(0..<10, id: \.self) { _ in
                            dappListItem()
                        }
                    }
                }
                .padding()
            }
            .navigationTitle("DApps")
            .toolbar {
                ToolbarItem(placement: .navigationBarTrailing) {
                    Button(action: {}) {
                        Image(systemName: "magnifyingglass")
                    }
                }
            }
        }
    }
    
    func dappCard(name: String, category: String, icon: String) -> some View {
        VStack {
            ZStack {
                RoundedRectangle(cornerRadius: 12)
                    .fill(Color.gray.opacity(0.2))
                    .frame(width: 100, height: 100)
                Text(icon)
                    .font(.largeTitle)
            }
            Text(name)
                .font(.subheadline)
                .fontWeight(.medium)
            Text(category)
                .font(.caption)
                .foregroundColor(.secondary)
        }
    }
    
    func dappListItem() -> some View {
        HStack(spacing: 12) {
            ZStack {
                Circle()
                    .fill(Color.gray.opacity(0.2))
                    .frame(width: 44, height: 44)
                Text("🪙")
            }
            
            VStack(alignment: .leading) {
                Text("DEX Protocol")
                    .font(.subheadline)
                Text("DeFi")
                    .font(.caption)
                    .foregroundColor(.secondary)
            }
            
            Spacer()
            
            Button(action: {}) {
                Text("Connect")
                    .font(.caption)
                    .padding(.horizontal, 12)
                    .padding(.vertical, 6)
                    .background(Color.orange)
                    .foregroundColor(.white)
                    .cornerRadius(8)
            }
        }
        .padding()
        .background(Color.gray.opacity(0.1))
        .cornerRadius(12)
    }
    
    var activityView: some View {
        NavigationView {
            ScrollView {
                VStack(spacing: 12) {
                    ForEach(0..<20, id: \.self) { _ in
                        activityItem()
                    }
                }
                .padding()
            }
            .navigationTitle("Activity")
        }
    }
    
    func activityItem() -> some View {
        HStack(spacing: 12) {
            Image(systemName: "arrow.up.circle.fill")
                .font(.title2)
                .foregroundColor(.orange)
            
            VStack(alignment: .leading, spacing: 2) {
                Text("Sent ETH")
                    .font(.subheadline)
                Text("To: 0x742d...f123")
                    .font(.caption)
                    .foregroundColor(.secondary)
            }
            
            Spacer()
            
            VStack(alignment: .trailing, spacing: 2) {
                Text("-0.5 ETH")
                    .font(.subheadline)
                    .foregroundColor(.red)
                Text("Just now")
                    .font(.caption)
                    .foregroundColor(.secondary)
            }
        }
        .padding()
        .background(Color.gray.opacity(0.1))
        .cornerRadius(12)
    }
}

struct TokenItem: Identifiable {
    let id = UUID()
    let symbol: String
    let name: String
    let balance: Double
    let value: Double
    let icon: String
}

#Preview {
    HomeScreen()
}

File: mobile_apps/android_app/TigerWallet/app/src/main/java/com/tigerwallet/app/ui/screens/HomeScreen.kt
package com.tigerwallet.app.ui.screens

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp

// Home Screen - Main Dashboard
class HomeScreen : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            MaterialTheme {
                HomeScreenContent()
            }
        }
    }
}

@OptIn(ExperimentalMaterialApi::class)
@Composable
fun HomeScreenContent() {
    var selectedTab by remember { mutableStateOf("Wallet") }
    val totalBalance = 12450.00

    val tokens = listOf(
        TokenData("ETH", "Ethereum", 4.2, 8400.00, "◈"),
        TokenData("USDT", "Tether", 2500.0, 2500.00, "₮"),
        TokenData("BNB", "BNB", 1.5, 450.00, "🟡"),
        TokenData("SOL", "Solana", 22.0, 1100.00, "☀️")
    )

    Scaffold(
        bottomBar = {
            BottomNavigation(
                selectedTab = selectedTab,
                onTabSelected = { selectedTab = it }
            )
        }
    ) { paddingValues ->
        LazyColumn(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
                .padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(20.dp)
        ) {
            // Balance Card
            item {
                Card(
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(16.dp),
                    elevation = CardDefaults.elevation(8.dp)
                ) {
                    Box(
                        modifier = Modifier
                            .fillMaxWidth()
                            .background(
                                Brush.linearGradient(
                                    colors = listOf(
                                        Color(0xFFFF6B35),
                                        Color(0xFFFF8C5A)
                                    )
                                )
                            )
                            .padding(24.dp)
                    ) {
                        Column(
                            horizontalAlignment = Alignment.CenterHorizontally,
                            modifier = Modifier.fillMaxWidth()
                        ) {
                            Text(
                                "Total Balance",
                                style = MaterialTheme.typography.subtitle1,
                                color = Color.White.copy(alpha = 0.8f)
                            )
                            Spacer(modifier = Modifier.height(8.dp))
                            Text(
                                "$${String.format("%.2f", totalBalance)}",
                                style = MaterialTheme.typography.h3,
                                fontWeight = FontWeight.Bold,
                                color = Color.White
                            )
                            Spacer(modifier = Modifier.height(20.dp))
                            Row(
                                horizontalArrangement = Arrangement.spacedBy(20.dp)
                            ) {
                                Button(
                                    onClick = { /* Navigate to Send */ },
                                    colors = ButtonDefaults.buttonColors(
                                        backgroundColor = Color.White,
                                        contentColor = Color(0xFFFF6B35)
                                    ),
                                    shape = RoundedCornerShape(20.dp)
                                ) {
                                    Icon(androidx.compose.material.icons.Icons.Default.ArrowUpward, contentDescription = null)
                                    Spacer(modifier = Modifier.width(4.dp))
                                    Text("Send")
                                }
                                OutlinedButton(
                                    onClick = { /* Navigate to Receive */ },
                                    colors = ButtonDefaults.outlinedButtonColors(
                                        contentColor = Color.White
                                    ),
                                    shape = RoundedCornerShape(20.dp)
                                ) {
                                    Icon(androidx.compose.material.icons.Icons.Default.ArrowDownward, contentDescription = null)
                                    Spacer(modifier = Modifier.width(4.dp))
                                    Text("Receive")
                                }
                            }
                        }
                    }
                }
            }

            // Quick Actions
            item {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceEvenly
                ) {
                    QuickActionButton(icon = "🔄", label = "Swap", color = Color.Blue)
                    QuickActionButton(icon = "🌉", label = "Bridge", color = Color.Magenta)
                    QuickActionButton(icon = "📈", label = "Stake", color = Color.Green)
                    QuickActionButton(icon = "🖼️", label = "NFTs", color = Color.Red)
                }
            }

            // Assets Header
            item {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text("Assets", style = MaterialTheme.typography.h6)
                    TextButton(onClick = { /* Add token */ }) {
                        Text("Add Token", color = Color(0xFFFF6B35))
                    }
                }
            }

            // Token List
            items(tokens) { token ->
                TokenRow(token = token)
            }
        }
    }
}

@Composable
fun QuickActionButton(icon: String, label: String, color: Color) {
    Column(
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Box(
            modifier = Modifier
                .size(50.dp)
                .clip(CircleShape)
                .background(color.copy(alpha = 0.2f)),
            contentAlignment = Alignment.Center
        ) {
            Text(icon, style = MaterialTheme.typography.h5)
        }
        Spacer(modifier = Modifier.height(4.dp))
        Text(label, style = MaterialTheme.typography.caption)
    }
}

@Composable
fun TokenRow(token: TokenData) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(12.dp)
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Icon
            Box(
                modifier = Modifier
                    .size(44.dp)
                    .clip(CircleShape)
                    .background(Color.Gray.copy(alpha = 0.2f)),
                contentAlignment = Alignment.Center
            ) {
                Text(token.icon, style = MaterialTheme.typography.title4)
            }
            
            Spacer(modifier = Modifier.width(12.dp))
            
            // Name & Symbol
            Column(modifier = Modifier.weight(1f)) {
                Text(token.name, style = MaterialTheme.typography.subtitle1, fontWeight = FontWeight.Medium)
                Text(token.symbol, style = MaterialTheme.typography.caption, color = MaterialTheme.colors.onSurfaceVariant)
            }
            
            // Balance & Value
            Column(horizontalAlignment = Alignment.End) {
                Text("${String.format("%.4f", token.balance)}", style = MaterialTheme.typography.subtitle1, fontWeight = FontWeight.Medium)
                Text("$${String.format("%.2f", token.value)}", style = MaterialTheme.typography.caption, color = MaterialTheme.colors.onSurfaceVariant)
            }
        }
    }
}

data class TokenData(
    val symbol: String,
    val name: String,
    val balance: Double,
    val value: Double,
    val icon: String
)

@Composable
fun BottomNavigation(
    selectedTab: String,
    onTabSelected: (String) -> Unit
) {
    val tabs = listOf("Wallet" to androidx.compose.material.icons.Icons.Default.Wallet, 
                      "DApps" to androidx.compose.material.icons.Icons.Default.Apps,
                      "Activity" to androidx.compose.material.icons.Icons.Default.History)

    BottomNavigation {
        tabs.forEach { (label, icon) ->
            BottomNavigationItem(
                selected = selectedTab == label,
                onClick = { onTabSelected(label) },
                icon = { Icon(icon, contentDescription = label) },
                label = { Text(label) }
            )
        }
    }
}
